src/qml/synchelper.cpp

```diff
@@ -118,24 +114,56 @@ void SyncHelper::slotSyncStatus(const QString &aProfileId, int aStatus,
         setLoading(true);
     }
 }
-
 void SyncHelper::slotProfileChanged(QString aProfileId, int aChangeType, QString aChangedProfile)
 {
     Q_UNUSED(aChangedProfile);
 
     if (aChangeType == Buteo::ProfileManager::PROFILE_REMOVED) {
-        QString profileNamePrefix = SocialSyncInterface::profileName(m_socialNetwork, m_dataType) + "-";
-        if (aProfileId.startsWith(profileNamePrefix)) {
+        if (profileIdMatches(aProfileId)) {
             emit profileDeleted();
         }
     }
 }
 
+void SyncHelper::refresh()
+{
+    refreshSyncProfiles();
+    checkCurrentRun();
+}
+
+void SyncHelper::refreshSyncProfiles()
+{
+    if (!m_complete) {
+        return;
+    }
+    QStringList syncProfiles;
+    Q_FOREACH (const QString &profileId, m_interface->syncProfilesByType(QStringLiteral("sync"))) {
```

Comment:
while this will be the case for images sync, it may not be for other types (e.g, backup storage etc)